### PR TITLE
add context to client data fetching page aiming to be easier to understand

### PR DESCRIPTION
**Pull request by eduarmreyes**
## Description

This Pull Request aims to add context about hydration and build and run time to be easier to understand particularly for beginners

## Related Issues

Addresses https://github.com/gatsbyjs/gatsby/issues/14938

### CODE REVIEW

File: docs/docs/client-data-fetching.md [hunk 1]
```diff
@@ -2,7 +2,17 @@
 title: Client data fetching
 ---
 
-Because a Gatsby site hydrates into a React app after loading statically, Gatsby is not just for static sites. You can fetch data dynamically on the client, as needed, as you would with any other React app.
+## Context
+
+In this article it is touched **how** to fetch data and **how** to handle that data using the plugin [`gatsby-source-graphql`](/packages/gatsby-source-graphql/).
```
Comment (marcysutton) [inline docs/docs/client-data-fetching.md]:
```suggestion
This article touches on _how to fetch_ data and _how to handle_ that data using the plugin [`gatsby-source-graphql`](/packages/gatsby-source-graphql/).
```

File: docs/docs/client-data-fetching.md [hunk 2]
```diff
@@ -2,7 +2,17 @@
...
+
+When this article says "hydration" it means that Gatsby through ReactJS already know the markup of your site because it was rendered in the server, so React will preserve the markup and attach event handlers to it, the result of this process allows you to have a really nice performant first-load time experience.
```
Comment (marcysutton) [inline docs/docs/client-data-fetching.md]:
```suggestion
When this article mentions [hydration](/docs/glossary#hydration), it means that Gatsby (through React.js) builds static files to render server-side. When a page loads and React's [client-side](/docs/glossary#client-side) code downloads and executes, it preserves the HTML markup built by Gatsby and turns the site into a full JavaScript web application that can manipulate the [DOM](/docs/glossary#dom). The result of this process creates fast loading pages and a nice user experience.
```

File: docs/docs/client-data-fetching.md [hunk 3]
```diff
@@ -2,7 +2,17 @@
...
+
+Build-time is useful when we know that the markup of our websites will not change or we do not really care if it changes, as mentioned before, the end-result of this process is a really fast first-load time experience as for run-time is appropriate for constantly changing markup like a chat or an email client web app.
```
Comment (marcysutton) [inline docs/docs/client-data-fetching.md]:
```suggestion
Compiling pages at [build-time](/docs/glossary#build) is useful when your website content will not change often, or triggering a build process to recompile is fine. However, some websites have more dynamic needs requiring [client-side](/docs/glossary#client-side) code to handle constantly changing content, like a chat widget or an email client web application.
```

File: docs/docs/client-data-fetching.md [hunk 4]
```diff
@@ -2,7 +2,17 @@
...
+
+We also talk about build-time which is defined for the process when Gatsby loads all the configuration and set-up everything to start up the website and run-time which is defined for the process when Gatsby has loaded everything up and the website is running in the server environment.
```
Comment (marcysutton) [inline docs/docs/client-data-fetching.md]: I think this paragraph could be removed, as it seems more likely to confuse than clarify things.

**Comment by m-allanson** [comment]
The failing test here seemed unrelated - I've restarted it to see if that fixes it.

**Comment by marcysutton** [comment]
I've pushed an update to finish this out, it should be good to go! thanks so much for your contribution @eduarmreyes, we appreciate it!

**Comment by eduarmreyes** [comment]
Thank you @marcysutton, I had the greatest experience contributing, was unable to push the latest review though but either way, I feel I did something, thank you for that. I will search for another issue easy to do and will try to contribute more.

**Pull request merged**